Task 1/Assets/Scripts/Camera/CameraController.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections;
+using UnityEngine;
+
+namespace StackApp.CameraController
+{
+    public class CameraController : MonoBehaviour
+    {
+        private Camera playerCamera;
+        private GameConfigurationDataWorld worldConfig;
+        private GameConfigurationDataBlock blockConfig;
+       
+        public void Initialization(GameConfigurationDataWorld worldConfiguration, GameConfigurationDataBlock blockConfiguration)
+        {
+            worldConfig = worldConfiguration;
+            blockConfig = blockConfiguration;
+
+            playerCamera = Camera.main;
+            playerCamera.transform.position = worldConfig.CameraStartPosition;
+            playerCamera.transform.rotation = Quaternion.Euler(worldConfig.CameraStartRotation);
+
+            EventController.OnGameRestarting += OnGameRestarting;
+            EventController.OnBlockRightPlacing += OnBlockRightPlacing;
+        }
+
+        private IEnumerator CameraMove(Vector3 target, float speed)
+        {
+            while (Vector3.Distance(playerCamera.transform.position, target) > float.Epsilon)
+            {
+                playerCamera.transform.position = Vector3.MoveTowards(playerCamera.transform.position, target, speed * Time.deltaTime);
+                yield return null;
+            }
+        }
```

Comment:
Is it possible, that you would do the second move before previous CameraMove has finished? With any different speeds? if so - would it work fine?